src/main/java/com/pinterest/secor/uploader/HadoopS3UploadManager.java

```diff
@@ -46,11 +46,13 @@ public HadoopS3UploadManager(SecorConfig config) {
     public Handle<?> upload(LogFilePath localPath) throws Exception {
         String prefix = FileUtil.getPrefix(localPath.getTopic(), mConfig);
         LogFilePath path = localPath.withPrefix(prefix);
+        path.setCustomTopicsNames(mConfig.getCustomTopicsNames());
```

Comment:
This (customTopicNames) should be added when the LogFilePath was created, not later.